grpc-gcp/src/main/java/com/google/cloud/grpc/multiendpoint/MultiEndpoint.java

```diff
@@ -233,9 +261,34 @@ private synchronized void updateCurrentEndpoint(Endpoint current, String newCurr
   }
 
   private synchronized void switchCurrentEndpoint() {
+    registerSwitch(currentId, switchTo);
     currentId = switchTo;
   }
 
+  private void registerSwitch(String fromId, String toId) {
+    if (toId.equals(fromId) || fromId == null) {
+      return;
+    }
+
+    int fromIdx = -1;
+    int toIdx = -1;
+    for (Endpoint e :endpointsMap.values()) {
+      if (e.getId().equals(fromId)) {
+        fromIdx = e.getPriority();
+      } else if (e.getId().equals(toId)) {
+        toIdx = e.getPriority();
+      }
+    }
+
+    if (fromIdx == -1) {
+      replaceCnt++;
+    } else if (fromIdx < toIdx) {
+      fallbackCnt++;
+    } else {
+      recoverCnt++;
```

Comment:
Just double check, when fromIdx == toIdx, the type is also recover?

Reply:
`fromIdx == toIdx` if and only if `toId.equals(fromId)` because the priority assigned to an endpoint comes from incrementing counter and is unique. And this condition is checked in the first line of this method.